telegram/ext/filters.py

```diff
@@ -182,8 +183,16 @@ def __init__(self, name: str = None, data_filter: bool = False):
         self._name = self.__class__.__name__ if name is None else name
         self._data_filter = data_filter
 
-    def check_update(self, update: Update) -> Optional[Union[bool, DataDict]]:  # skipcq: PYL-R0201
-        """Checks if the specified update is a message."""
+    def check_update(self, update: Update) -> bool:  # skipcq: PYL-R0201
```

Comment:
the type hint change should be reverted. The subclasses of `BaseFilter`s `check_update` can return `DataDict` based on `data_filter` property. That's why pylint also complained in `UpdateFilter` and `MessageFilter`

Reply:
@harshil21 Are you sure? the base method can only return the True of False 🤔 The methods which overrides this one can return the `DataDict` 🤔 
<img width="556" alt="Screenshot 2022-10-01 at 16 21 31" src="https://user-images.githubusercontent.com/2768101/193414037-6cf749c0-2787-4320-8b79-92516fb3d9a5.png">

Reply:
imo, base method should have same sig as children (if possible). Let's see what others think

Reply:
@harshil21 I reverted the change for now, we can do it in another PR to get others opinions on this specific case later 👍

Reply:
I agree with harshil here.